atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/cleaner/AsyncPuncher.java

```diff
@@ -47,7 +47,7 @@ public static AsyncPuncher create(Puncher delegate, long interval, LongSupplier
         return asyncPuncher;
     }
 
-    private static final ScheduledExecutorService executor =
+    private final ScheduledExecutorService executor =
             PTExecutors.newSingleThreadScheduledExecutor(new NamedThreadFactory("puncher", true /* daemon */));
```

Comment:
how many instances of this class are created? would it be possible to  inject the executor and shut down when the parent class shuts down?